internal/common/conversion/type_conversion.go

```diff
@@ -86,3 +96,11 @@ func NilForUnknown[T any](primitiveAttr TFPrimitiveType, value *T) *T {
 	}
 	return value
 }
+
+func NilForUnknownOrEmpty(primitiveAttr types.String) *string {
```

Comment:
should we add in the func name "string" as it can also be used for ints or other types?

Reply:
I like it, changed in 4b16f455ea32f9f74f93c78d8fcbc9c889589139